packages/contracts/contracts/APM/PluginRepo.sol

```diff
@@ -0,0 +1,226 @@
+/*
+ * SPDX-License-Identifier:    MIT
+ */
+
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "../core/acl/ACL.sol";
+import "../core/erc165/AdaptiveERC165.sol";
+import "../APM/IPluginFactory.sol";
+import "./IPluginRepo.sol";
+
+/// @title The repository contract required for managing and publishing different version of a plugin within the Aragon DAO framework
+/// @author Aragon Association - 2020 - 2022
+contract PluginRepo is IPluginRepo, Initializable, UUPSUpgradeable, ACL, AdaptiveERC165 {
+    bytes32 public constant CREATE_VERSION_ROLE = keccak256("CREATE_VERSION_ROLE");
+    bytes32 public constant UPGRADE_ROLE = keccak256("UPGRADE_ROLE");
+
+    /// @notice Thrown version bump is invalid
+    error InvalidBump();
+
+    /// @notice Thrown if version is invalid
+    error InvalidVersion();
+
+    /// @notice Thrown if version does not exist
+    error VersionDoesNotExist();
+
+    /// @notice Thrown if contract does not have IPluginFactory interface
+    error InvalidPluginInterface();
+
+    /// @notice Thrown if address is not a PluginFactory contract
+    error InvalidPluginContract();
+
+    /// @notice Thrown if address is not a contract
+    error InvalidContract();
```

Comment:
Be more verbose by adding more information to the error and use this information in the tests.
Which bump is invalid?
What version is invalid?
What's the interface id that is invalid?